include/boost/geometry/extensions/gis/io/geojson/geojson_writer.hpp

```diff
@@ -0,0 +1,188 @@
+// Boost.Geometry
+
+// Copyright (c) 2024 Barend Gehrels, Amsterdam, the Netherlands.
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_EXTENSIONS_GIS_IO_GEOJSON_WRITER_HPP
+#define BOOST_GEOMETRY_EXTENSIONS_GIS_IO_GEOJSON_WRITER_HPP
+
+#include <boost/geometry/core/access.hpp>
+#include <boost/geometry/io/dsv/write.hpp>
+
+#include <ostream>
+#include <string>
+
+namespace boost { namespace geometry
+{
+
+#ifndef DOXYGEN_NO_DISPATCH
+namespace dispatch
+{
+
+template <typename GeometryTag>
+struct geojson_feature_type
+{
+    BOOST_GEOMETRY_STATIC_ASSERT_FALSE("Not or not yet implemented for this Geometry type.",
+                                       GeometryTag);
+};
+
+template <>
+struct geojson_feature_type<point_tag> { static inline const char* apply() { return "Point"; } };
+
+template <>
+struct geojson_feature_type<segment_tag>  { static inline const char* apply() { return "LineString"; } };
+
+template <>
+struct geojson_feature_type<ring_tag>  { static inline const char* apply() { return "Polygon"; } };
+
+template <>
+struct geojson_feature_type<linestring_tag>  { static inline const char* apply() { return "LineString"; } };
+
+template <>
+struct geojson_feature_type<polygon_tag>  { static inline const char* apply() { return "Polygon"; } };
+
+template <>
+struct geojson_feature_type<multi_polygon_tag>  { static inline const char* apply() { return "MultiPolygon"; } };
+
+} // namespace dispatch
+#endif
+
+/*!
+\brief Helper class to create geojson output
+*/
+struct geojson_writer
+{
+
+  private:
+    std::ostream& m_out;
+
+    static const char* comma_space() { return ", "; }
+    static const char* colon() { return ": "; }
+    static const char* gtag() { return "\"geometry\""; }
+    static const char* ttag() { return "\"type\""; }
+    static const char* ctag() { return "\"coordinates\""; }
+    static char comma() { return ','; }
+    static char quote() { return '"'; }
+    static char cbopen() { return '{'; }
+    static char cbclose() { return '}'; }
```

Comment:
Is there a motivation for these functions on lines  62, 63, 67-70 if they do not maintain/validate any state, rather than writing the chars/strings explicitly into the output stream where used?

Reply:
There is a weak motivation coming from another code base where I'm working on, where it is the preferred way not to use constants or literals directly in the code.

But indeed - no need to adhere to that here. I'll update it, thanks.

Reply:
✅ all updated and optimized a bit using `R("` etc.